config/dynamic.c

```diff
@@ -601,6 +605,21 @@ run_dynamic_config(struct gk_config *gk_conf, struct gt_config *gt_conf,
 	if (ret < 0)
 		goto put_gk_gt_config;
 
+	ret = chown(dy_conf->server_path, net_conf->pw_uid, net_conf->pw_gid);
+	if (ret < 0) {
+		DYC_LOG(ERR, "Failed to change the owner of the file (%s) to user with uid %u and gid %u - %s\n",
+			dy_conf->server_path, net_conf->pw_uid,
+			net_conf->pw_gid, strerror(errno));
+		goto put_gk_gt_config;
+	}
+
+	ret = chmod(dy_conf->server_path, mode);
```

Comment:
Can't we call `fchmod()` on `dy_conf->sock_fd` instead of calling `chmod()`? This avoids rare file-system-related race conditions.